array_methods_lab_REDUCE.js

```diff
@@ -0,0 +1,299 @@
+// # Exercises
+
+// 1. Given an array of numbers, write a function that returns a new array where every element is incremented by 10.
+
+// ```
+// Input: [1,2,3,4,5]
+// Output: [11,12,13,14,15]
+// ```
+const incTen = (arr) => {
+    return arr.reduce((acc, el) => {
+        acc.push(el+10);
+        return acc;
+    }, [])
+}
+console.log(incTen([1,2,3,4,5]));
+
+
+// 2. Given an array, return a new array where every non-string element is converted to an empty string.
+
+// ```
+// Input: ["a", 123, "b", "c", {name: "cat"}]
+// Output: ["a", "", "b", "c", ""]
+// ```
+const emptyStr = (arr) => {
+    return arr.reduce((acc, el) => {
+        if(typeof el === 'string') {
+            acc.push(el);
+        } else {
+            acc.push('');
+        }
+        return acc;
+    }, [])
+}
+console.log(emptyStr(["a", 123, "b", "c", {name: "cat"}]));
+
+// 3. Write a function that returns only the even elements in an array of numbers.
+
+// ```
+// Input: [1,2,3,4,5]
+// Output: [2,4]
+// ```
+const evenElements = (arr) => {
+    return arr.reduce((acc,el) => {
+        if(el % 2 === 0){
+            acc.push(el);
+        }
+        return acc;
+    },[])
+}
+console.log(evenElements([1,2,3,4,5]));
+
+// 4. Write a function that checks if all elements in the array are the same.
+
+// ```
+// Input: [1,1,1,1,1]
+// Output: true
+
+// Input: [1,1,1,1,2]
+// Output: false
+// ```
+const arrEveryReduce = (arr) => {
+    return arr.reduce((acc, el) => {
```

Comment:
You want to keep track if its true or false in a different way. Let's say your last element is the same as arr[0], but the previous one wasn't. This would give you back true, even though they're all not the same.